### ISSUE TITLE
Compliance Property: Digital Proof of Address (Registration number) 

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#763]
## 🚀 Feature Summary

Implement a digital verification system that validates a property’s official registration number and proof of address during host onboarding and listing creation. The system cross-checks submitted registration details against government or municipal databases to ensure the property is legally registered for short-term rental operations. This helps prevent fraudulent listings, supports regulatory compliance, and protects the platform in regulated markets. The feature strengthens trust with authorities, enterprise partners, and guests while enabling scalable compliance automation.

- [ ] Analyze current v1 setup
- [ ] Compliance: Digital Proof of Address: Verify your listing with registration number
- [ ] ...

---

## Problem Statement  
In many cities and countries, short-term rentals require an official registration number and verified property address. Without structured validation, unregistered or fraudulent listings can be published, exposing the platform to regulatory fines, legal disputes, reputational damage, and forced delistings.

Hosts may unknowingly operate non-compliant properties, guests may book illegal listings, and the platform risks non-compliance with local authorities.

> The core problem is lack of automated, scalable compliance validation at the property level.

---

## Desired Outcome  
All active listings have a verified registration number and validated proof of address where required by local law. Non-compliant properties are flagged or prevented from going live. The platform can demonstrate regulatory compliance to municipalities and enterprise partners.

---

## Target Users  

Who benefits from this feature?

- [x] Guests / Customers  
- [x] Hosts / Property Managers / Enterprise  
- [x] Admin / Internal users  
- [ ] Developers  
- [x] Other: Regulators / Municipal Authorities  

---

## Business & User Impact  

**Impact level:**  
- [ ] Low (nice-to-have)  
- [ ] Medium (improves workflows)  
- [ ] High (unblocks key use cases)  
- [x] Critical (strategic / revenue-impacting)  

**User reach:**  
- [ ] Few users  
- [ ] Many users  
- [x] All users  
- [ ] Unknown  

---

## Proposed Solution (Optional)  

During onboarding or listing creation:

- Require hosts to submit a registration number and proof of address document.  
- Validate format rules per jurisdiction.  
- Where available, cross-check against official municipal or national databases.  
- Flag discrepancies for manual review.  
- Prevent publication until verification is complete (if required by jurisdiction).

---

## Alternatives Considered  

- Manual verification by compliance team only (not scalable).  
- Host self-declaration without validation (high regulatory risk).  
- Post-publication audits (reactive rather than preventative).

---

## Constraints & Considerations  

- **Performance:** Verification should not significantly delay onboarding.  
- **Security:** Documents must be encrypted in transit and at rest.  
- **Privacy:** Data handling must comply with regulations such as the General Data Protection Regulation (GDPR).  
- **Accessibility:** Clear guidance for hosts in different jurisdictions.  
- **Backward compatibility:** Existing listings may require phased verification.  
- **External dependencies:** Reliance on municipal or government APIs (availability varies by country).

---

## Acceptance Criteria (Optional but Encouraged)  

- [ ] Given a required jurisdiction, when a host submits a listing without a registration number, then publication is blocked.  
- [ ] Given an invalid registration format, when submitted, then the system returns a validation error.  
- [ ] Given a valid registration number, when verified successfully, then the listing status updates to “Compliant.”  
- [ ] Given verification failure, then the listing is flagged for review.
